src/probnum/diffeq/perturbedsolvers/perturbedstepsolver.py

```diff
@@ -0,0 +1,91 @@
+"""ODE solver as proposed by Abdulle and Garegnani."""
+import numpy as np
+
+from probnum import diffeq, randvars
+from probnum.diffeq import wrappedscipysolver
+from probnum.diffeq.perturbedsolvers import (
+    _perturbation_functions,
+    perturbedstepsolution,
+)
+from probnum.type import FloatArgType
+
+
+class PerturbedStepSolver(diffeq.ODESolver):
+    """ODE Solver based on Scipy that introduces uncertainty by perturbing the time-
+    steps."""
+
+    # pylint: disable=maybe-no-member
+    def __init__(
+        self,
+        solver: wrappedscipysolver,
+        noise_scale: np.array,
+        perturb_function,
+        random_state=None,
+    ):
+        def perturbation_step(step):
+            return perturb_function(
+                step=step,
+                solver_order=solver.order,
+                noise_scale=noise_scale,
+                random_state=random_state,
```

Comment:
it might be good to wrap `random_state` through `utils.as_random_state` before this call. Otherwise, if someone accidentally passes an integer seed instead of a proper random state object, the perturbation would be constant?